### PR TITLE
improve build action

### PR DESCRIPTION
**Pull request by KeisukeShima**
## What kinds of PR

- [ ] New feature
- [x] Improve feature
- [ ] Bug fix

## What is this PR
I evaluated the following three types of Actions and suggest to use `actions-ros + setup-ros on ubuntu:focal`.
- actions-ros + setup-ros on ubuntu:focal
  - It works well. As Faster as old one.
- actions-ros + setup-ros on osrf/ros:foxy-desktop
  - setup-ros conflicts `osrf/ros:foxy-desktop`. See [here](https://github.com/tier4/obstacle_stop_planner_refine/runs/2027140688?check_suite_focus=true).
- industrial_ci
  -This is not suitable for this project because it doesn't have colcon build options. So I cannot use `--packages-up-to`.

## How to check this PR
See result of GitHub Actions.

### CODE REVIEW

File: .github/workflows/action-ros-ci-with-setup-ros.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Build and test(actions-ros + setup-ros on ubuntu:focal)
```
Comment (kmiya) [inline .github/workflows/action-ros-ci-with-setup-ros.yml]: nits: It would be better to insert a space before a paren, i.e., `test (actions` [(ref)](https://english.stackexchange.com/questions/5987/is-there-any-rule-for-the-placement-of-space-after-and-before-parentheses)

File: .github/workflows/action-ros-ci-with-setup-ros.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    container: ubuntu:focal
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+
+      - name: setup ROS environment
```
Comment (kmiya) [inline .github/workflows/action-ros-ci-with-setup-ros.yml]: nits: It would be better to begin with upper case (`Setup`)

File: .github/workflows/action-ros-ci-with-setup-ros.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        uses: ros-tooling/setup-ros@v0.1
+        with:
+          required-ros-distributions: foxy
+
+      - name: action-ros-ci
+        uses: ros-tooling/action-ros-ci@v0.1
+        with:
+          package-name: obstacle_stop_planner_refine
+          target-ros2-distro: foxy
+          vcs-repo-file-url: build_depends.repos
```
Comment (kenji-miyake) [inline .github/workflows/action-ros-ci-with-setup-ros.yml]: [Question] Could you explain what commands are executed with these settings for my understanding? (for example by referring to the action's code)

File: obstacle_stop_planner_refine/package.xml [hunk 4]
```diff
@@ -8,7 +8,7 @@
 
   <license>Apache 2.0</license>
 
-  <buildtool_depend>ament_cmake</buildtool_depend>
+  <buildtool_depend>ament_cmake_auto</buildtool_depend>
```
Comment (kenji-miyake) [inline obstacle_stop_planner_refine/package.xml:11]: [Question] Why did you change this? Some errors have occurred?

File: .github/workflows/action-ros-ci-with-setup-ros.yml (hunk 3 above)
Comment (KeisukeShima) [reply .github/workflows/action-ros-ci-with-setup-ros.yml]: actions-ros performs `colcon build + colcon test + colcon lcov-result + colcon coveragepy-result`
You can check commands here: <https://github.com/tier4/obstacle_stop_planner_refine/runs/2029944928?check_suite_focus=true>

image

File: .github/workflows/action-ros-ci-with-setup-ros.yml (hunk 1 above)
Comment (KeisukeShima) [reply .github/workflows/action-ros-ci-with-setup-ros.yml]: Thanks! Fixed e391c38

File: .github/workflows/action-ros-ci-with-setup-ros.yml (hunk 2 above)
Comment (KeisukeShima) [reply .github/workflows/action-ros-ci-with-setup-ros.yml]: Thanks! Fixed e391c38

File: obstacle_stop_planner_refine/package.xml (hunk 4 above)
Comment (KeisukeShima) [reply obstacle_stop_planner_refine/package.xml:11]: I got error when I use `ament_cmake`. Maybe `rosdep` needs `ament_cmake_auto` to find package.
In other environment, `ament_cmake_auto` has already installed so that there are no errors. But I think using `ament_cmake_auto` is better to avoid errors.

https://github.com/tier4/obstacle_stop_planner_refine/runs/2021744768?check_suite_focus=true

```
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
  Starting function 'build_target_workspace'
  Starting >>> obstacle_stop_planner_refine
  --- stderr: obstacle_stop_planner_refine
  CMake Error at CMakeLists.txt:14 (find_package):
    By not providing "Findament_cmake_auto.cmake" in CMAKE_MODULE_PATH this
    project has asked CMake to find a package configuration file provided by
    "ament_cmake_auto", but CMake did not find one.
  
    Could not find a package configuration file provided by "ament_cmake_auto"
    with any of the following names:
  
      ament_cmake_autoConfig.cmake
      ament_cmake_auto-config.cmake
  
    Add the installation prefix of "ament_cmake_auto" to CMAKE_PREFIX_PATH or
    set "ament_cmake_auto_DIR" to a directory containing one of the above
    files.  If "ament_cmake_auto" provides a separate development package or
    SDK, be sure it has been installed.
  
  
  ---
```

File: .github/workflows/action-ros-ci-with-setup-ros.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        uses: ros-tooling/setup-ros@v0.1
+        with:
+          required-ros-distributions: foxy
+
+      - name: Run action-ros-ci
+        uses: ros-tooling/action-ros-ci@v0.1
+        with:
+          package-name: obstacle_stop_planner_refine
+          target-ros2-distro: foxy
+          vcs-repo-file-url: build_depends.repos
```
Comment (kenji-miyake) [inline .github/workflows/action-ros-ci-with-setup-ros.yml]: @KeisukeShima Should we keep the existing `build_and_test.yml` or will replace it with this?

Comment (KeisukeShima) [reply .github/workflows/action-ros-ci-with-setup-ros.yml]: @kenji-miyake I'd like to replace old file with this.

Comment (kenji-miyake) [reply .github/workflows/action-ros-ci-with-setup-ros.yml]: @KeisukeShima Then please remove it? :thinking:

Comment (KeisukeShima) [reply .github/workflows/action-ros-ci-with-setup-ros.yml]: Done! ee668db

**Review by kenji-miyake — DISMISSED** [review]

File: .github/workflows/action-ros-ci-with-setup-ros.yml (hunk 5 above)
Comment (kenji-miyake) [reply .github/workflows/action-ros-ci-with-setup-ros.yml]: @KeisukeShima By the way, the file name `action-ros-ci-with-setup-ros.yml` is okay for you? I feel renaming this to `build_and_test.yml` is better.

Comment (KeisukeShima) [reply .github/workflows/action-ros-ci-with-setup-ros.yml]: @kenji-miyake It's better. Renaming c93c459

**Review by kenji-miyake — APPROVED** [review]
